frontend/components/badge-examples.js

```diff
@@ -0,0 +1,91 @@
+import url from 'url';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function resolveUri (uri, baseUri, options) {
+  const { longCache } = options || {};
+  const resolved = baseUri ? url.resolve(baseUri, uri) : uri;
+  if (longCache) {
+    const parsed = url.parse(resolved, true);
+    parsed.query.maxAge = '2592000';
+    // url.format will re-format `query`, but only When `search` is deleted.
+    delete parsed.search;
+    return url.format(parsed);
+  } else {
+    return resolved;
+  }
+}
+
+const Badge = ({ title, previewUri, exampleUri, documentation, baseUri, isProductionBuild, onClick }) => {
+  const handleClick = () => onClick({ title, previewUri, exampleUri, documentation });
+
+  const previewImage = previewUri
+    ? (<img onClick={handleClick} src={resolveUri(previewUri, baseUri, { longCache: isProductionBuild } )} alt="" />)
+    : '\u00a0'; // non-breaking space
+  const resolvedExampleUri = resolveUri(exampleUri || previewUri, baseUri || 'https://img.shields.io/', { longCache: false });
+
+  return (
+    <tr><th onClick={handleClick}>{ title }:</th>
+      <td>{ previewImage }</td>
+      <td><code onClick={handleClick}>{ resolvedExampleUri }</code></td>
+    </tr>
+  );
+};
+Badge.propTypes = {
+  title: PropTypes.string.isRequired,
+  previewUri: PropTypes.string,
+  exampleUri: PropTypes.string,
+  documentation: PropTypes.string,
+  baseUri: PropTypes.string.isRequired,
+  isProductionBuild: PropTypes.bool.isRequired,
+  onClick: PropTypes.func.isRequired,
+};
```

Comment:
I feel like we should avoid propTypes (which only does runtime type checking and can't easily be statically analyzed) and instead use Flow.